src/main/java/seedu/task/logic/LogicManager.java

```diff
@@ -20,22 +21,34 @@
 
     private final Model model;
     private final Parser parser;
+    private UndoableCommand previousCommand;
 
     public LogicManager(Model model, Storage storage) {
         this.model = model;
         this.parser = new Parser();
+        this.previousCommand = null;
     }
 
     @Override
     public CommandResult execute(String commandText) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         Command command = parser.parseCommand(commandText);
-        command.setData(model);
-        return command.execute();
+        command.setData(model, previousCommand);
```

Comment:
Passing the previous command to the current command. Only Undo needs it, but there's no cleaner way to pass in the data.